### ISSUE TITLE
Is this library still actively maintained?

### ISSUE THREAD

**Issue by JonathonRP** [micha-lmxt/svelte-motion#36]

**Comment by enisbudancamanak** [comment]
It's so sad, that this powerful library is not up-to-date with the current SvelteKit Version, what can we do about that?

**Comment by JonathonRP** [comment]
> It's so sad, that this powerful library is not up-to-date with the current SvelteKit Version, what can we do about that?

I was hoping to have the current owner migrate to me otherwise I'll have to make another separate repo. I wanted to submit PRs but evidently the owner doesn't watch this repo and so I doubt there would be any merging even if I create a good PR with updates...

**Comment by enisbudancamanak** [comment]
Yeah I think he doesn't look over this repo anymore.. unfortunately. Would you do another repo? And is it fine with the rights to do that? I would be a big fan of this library to be still maintained. You have my support! :)

**Comment by enisbudancamanak** [comment]
Maybe it would be possible to send him a email or something.. i want this library so bad.

**Comment by JonathonRP** [comment]
> Yeah I think he doesn't look over this repo anymore.. unfortunately. Would you do another repo? And is it fine with the rights to do that? I would be a big fan of this library to be still maintained. You have my support! :)

That was one of the main reasons I reached out, but not super sure what to do with no response. Now most can be achieved with transitions and view transitions API now but still I used this framer motion library and like the declarative nature. And would like to make components that wrap with popmotion and svelte transitions and view transitions API

**Comment by JonathonRP** [comment]
> Maybe it would be possible to send him a email or something.. i want this library so bad.

That's a good idea, I'll try to email later and see if I get a response 👍

**Comment by enisbudancamanak** [comment]
Yes, I really fell in love with the declarative way of animations. I think svelte-motion is the only one who achieved it so powerful! 

Would really appreciate it, if u can keep me updated.. you have my support by the way. I would be down to support you with my contributions.

**Comment by fmaiabatista** [comment]
I'd like to follow updates here as well! :)

**Comment by JonathonRP** [comment]
So far I've reached out by email and have not gotten a response back yet.

**Comment by jbirtel** [comment]
> > It's so sad, that this powerful library is not up-to-date with the current SvelteKit Version, what can we do about that?
> 
> I was hoping to have the current owner migrate to me otherwise I'll have to make another separate repo. I wanted to submit PRs but evidently the owner doesn't watch this repo and so I doubt there would be any merging even if I create a good PR with updates...

I think he does reply on PRs, at least he replied to mine updating all the dependencies. So, I would suggest that if you have some good updates, then just try and drop a PR.

**Comment by JonathonRP** [comment]
> > > It's so sad, that this powerful library is not up-to-date with the current SvelteKit Version, what can we do about that?
> > 
> > 
> > I was hoping to have the current owner migrate to me otherwise I'll have to make another separate repo. I wanted to submit PRs but evidently the owner doesn't watch this repo and so I doubt there would be any merging even if I create a good PR with updates...
> 
> I think he does reply on PRs, at least he replied to mine updating all the dependencies. So, I would suggest that if you have some good updates, then just try and drop a PR.

awesome, thank you! I might, the main idea I had was to update to support svelte 5 which the component type needs to change to support svelte 4 and 5. also there is a bug with AnimatePresence with it's type declaration.

**Comment by jbirtel** [comment]
So, my PR is merged. I guess simply try to open one yourself :)

**Comment by JonathonRP** [comment]
Update: I will be opening PR soon with some of my ideas hopefully it gets merged 👍

**Comment by JonathonRP** [comment]
Update: sent PR with the minor changes I mentioned (component type and AnimatePresence bug fix)

**Comment by ermincelikovic** [comment]
The library is great, but I feel like the updates from the original author are quite slow. I get it, life happens - but @JonathonRP seems to be keen on keeping it alive?

@JonathonRP - would you consider re-publishing it under another name? rp-motion? :)

**Comment by JonathonRP** [comment]
> The library is great, but I feel like the updates from the original author are quite slow. I got it, life happens - but @JonathonRP seems to be keen to keep it alive?
> 
> @JonathonRP - would you consider re-publishing it under another name? rp-motion? :) 

I might, one day but I don't want to take it from the current author. He hasn't experienced interest in letting me fork and take ownership so I'm just making contributions and have a port to typescript PR and plans to bring it more inline with modern features and implementation of framer motion

**Comment by didier** [comment]
Whatever happens I'd love to be able to use this in my projects again. @micha-lmxt Please consider allowing @JonathonRP to fork or transfer to them if you are no longer interested in maintaining - no hard feelings :)

**Comment by cliffordkleinsr** [comment]
Anyone interested in porting this project over to svelte5/ kit2?

**Comment by JonathonRP** [comment]
> Anyone interested in porting this project over to svelte5/ kit2?

I'm doing that, and while porting to typescript btw

**Comment by didier** [comment]
@JonathonRP anywhere we could track progress and/or assist in any way?

**Comment by JonathonRP** [comment]
> @JonathonRP anywhere we could track progress and/or assist in any way?

https://github.com/micha-lmxt/svelte-motion/pull/47 is the pull request, been doing my best with commit messages to track progress. I have some changes not completely tested on branch right now but I'll be checking them in soon. Could use help testing, last test the functionality and animations were not working but I think I finally figured out why, but unable to test with my project that I've been testing with. That's actually the reason for the slowness in progress.


Note: I have aspersions to update and add features that are missing in this lib from original as original is still maintained, and also looking at using svelte built in transitions/ animations down the road. 
This typescript port begins taking steps in that direction.

**Comment by JonathonRP** [comment]
> > @JonathonRP anywhere we could track progress and/or assist in any way?
> 
> #47 is the pull request, been doing my best with commit messages to track progress. I have some changes not completely tested on branch right now but I'll be checking them in soon. Could use help testing, last test the functionality and animations were not working but I think I finally figured out why, but unable to test with my project that I've been testing with. That's actually the reason for the slowness in progress.
> 
> Note: I have aspersions to update and add features that are missing in this lib from original as original is still maintained, and also looking at using svelte built in transitions/ animations down the road. This typescript port begins taking steps in that direction.

changes are pushed, the main thing that could help speed process is having others test that functionality didn't change

**Comment by cliffordkleinsr** [comment]
How can I  help. Also will this pr be merged? @JonathonRP

**Comment by JonathonRP** [comment]
> How can I help. Also will this pr be merged? @JonathonRP

@cliffordkleinsr, if you are able to test functionality and contribute to PR changes that would be great help. the current PR is just to port to typescript and not change or break any current functionality.
my current understanding is it will get merged by current maintainer, I asked a question in issues here if he would accept a typescript port PR, he said yes. The PR is currently draft because during my testing functionality was broken but I think I fixed it, just need the latest changes tested to make sure that functionality is truly working and unchanged.

**Comment by ermincelikovic** [comment]
I honestly feel like it does not make much sense to wait on current maintainer - he's been busy/quiet for past couple of months (I also see that he became inactive on twitter).

It would totally make sense to fork this into a new library. I feel like svelte-motion is a bit confusing name to begin with (I know it comes from Framer motion), as svelte itself has a svelte/motion module.

**Comment by JonathonRP** [comment]
> I honestly feel like it does not make much sense to wait on current maintainer - he's been busy/quiet for past couple of months (I also see that he became inactive on twitter).
> 
> It would totally make sense to fork this into a new library. I feel like svelte-motion is a bit confusing name to begin with (I know it comes from Framer motion), as svelte itself has a svelte/motion module.

I may fork if he doesn't merge but I wanted to have a resource for learning and self taught to learn about animation which is made easier by using typescript.

@ermincelikovic 
What name do you think would be good for a fork?

**Comment by ermincelikovic** [comment]
I don't know, some examples
- **rp-motion** (gotta give yourself some credit)
- **declarative-motion** (still motion, but not to be confused with svelte/motion)
- **motion-kit** or **motionkit** (available on npm)
- **svelte-moves** - a bit funky :)

**Comment by JonathonRP** [comment]
> I don't know, some examples
> - **rp-motion** (gotta give yourself some credit)
> - **declarative-motion** (still motion, but not to be confused with svelte/motion)
> - **motion-kit** or **motionkit** (available on npm)
> - **svelte-moves** - a bit funky :)

@ermincelikovic
I was thinking
- **remotion**
- **svelte/animation** or **svelte/animations**
- **motion-start**

**Comment by ermincelikovic** [comment]
> > I don't know, some examples
> > 
> > * **rp-motion** (gotta give yourself some credit)
> > * **declarative-motion** (still motion, but not to be confused with svelte/motion)
> > * **motion-kit** or **motionkit** (available on npm)
> > * **svelte-moves** - a bit funky :)
> 
> @ermincelikovic I was thinking
> 
> * **remotion**
> * **svelte/animation** or **svelte/animations**
> * **motion-start**

**motion-start** is great imho!

remotion is a product (google it) - would be easily mistaken
svelte/animation could be mistaken for svelte/animate

**Comment by JonathonRP** [comment]
> > > I don't know, some examples
> > > 
> > > * **rp-motion** (gotta give yourself some credit)
> > > * **declarative-motion** (still motion, but not to be confused with svelte/motion)
> > > * **motion-kit** or **motionkit** (available on npm)
> > > * **svelte-moves** - a bit funky :)
> > 
> > @ermincelikovic I was thinking
> > 
> > * **remotion**
> > * **svelte/animation** or **svelte/animations**
> > * **motion-start**
> 
> **motion-start** is great imho!
> 
> remotion is a product (google it) - would be easily mistaken
> svelte/animation could be mistaken for svelte/animate

Yeah I was actually back and forth between **remotion** and (**re-motion** - little play on my name Reese) but there's also something really nice about **motion-start**

**Comment by cliffordkleinsr** [comment]
svelte-framer
svelte-framerMotion
svelterMotion
@ermincelikovic @JonathonRP

**Comment by JonathonRP** [comment]
> svelte-framer
> svelte-framerMotion
> svelterMotion
> @ermincelikovic @JonathonRP 

@ermincelikovic @cliffordkleinsr 
I like **svelteMotion**.
I thought of some others too,
- **svelte/motion-start**
- **svelte/framer-motion**

**Comment by cliffordkleinsr** [comment]
+1 for `svelte/framer-motion`
Honestly  have been itching to port the library into svelte5

**Comment by ermincelikovic** [comment]
Please no svelteMotion, it will add even more to the confusion of svelte-motion, svelte/motion 😅
svelte-framer-motion is not bad, but it has nothing to do with framer.

At the end of the day, it does not matter as long as we get to use it :)

**Comment by JonathonRP** [comment]
> +1 for `svelte/framer-motion`
> Honestly  have been itching to port the library into svelte5

Checking on npm packaging looks like I can't have **svelte/framer-motion**, because npm only allows **@org/package name** so it would have to be @svelte/framer-motion which I don't think svelte maintainers would appreciate... So it would have to be **motion-start** (framer-motion is the package) and (svelte-framer-motion seems a little too verbose) I could be swayed to **svelte-motion-start**

@ermincelikovic @cliffordkleinsr

**Comment by cliffordkleinsr** [comment]
Welp, I guess that'll have to do. Can we register it? @JonathonRP

**Comment by JonathonRP** [comment]
> Welp, I guess that'll have to do. Can we register it? @JonathonRP 

I will try to figureout how

**Comment by cliffordkleinsr** [comment]
Will also check tomorrow . Out of the office IRL

**Comment by JonathonRP** [comment]
> Will also check tomorrow . Out of the office IRL

@cliffordkleinsr,
Which name are you registering tomorrow,
**svelte-motion-start** or **motion-start**

**Comment by ermincelikovic** [comment]
I vote for motion-start

**Comment by cliffordkleinsr** [comment]
> I vote for motion-start

This

**Comment by JonathonRP** [comment]
@cliffordkleinsr, @ermincelikovic btw the typescript port is basically ready aside from testing that no functionality was added or lost.

**Comment by cliffordkleinsr** [comment]
will try and do some tests tomorrow. Which org should we register for the package?
Should also have a discord or something.

**Comment by JonathonRP** [comment]
> will try and do some tests tomorrow. Which org should we register for the package? Should also have a discord or something.

I wasn't going to us an org for the package...
if you want to setup discord server that is fine.

Also we should use changesets with this new package - this should have necessary information to add changesets to the code base and typescript port code base.

@cliffordkleinsr

**Comment by cliffordkleinsr** [comment]
Will do then. Let's start with the repo forking into the new  `motion-start` lib. Will try and register the npm tomorrow  and get the discord up.  initialize the library with the changesets workflow  probably  then we can start porting changes.

**Comment by JonathonRP** [comment]
@cliffordkleinsr, are you forking or do you want me to fork?

**Comment by cliffordkleinsr** [comment]
> @cliffordkleinsr, are you forking or do you want me to fork?

Please fork @JonathonRP . Let's get this show on the road

**Comment by JonathonRP** [comment]
@cliffordkleinsr, I already have a fork that was used to create PRs. https://github.com/JonathonRP/svelte-motion. should I change repo name and package.json name to **motion-start** or make it a template repo and make another with mentioned name?

**Comment by JonathonRP** [comment]
@cliffordkleinsr, @ermincelikovic  please find new repo - https://github.com/JonathonRP/motion-start
typescript port and changesets are in codebase and name svelte-motion is now reserved as it's published on npm fyi

please test typescript and note it is 0.0.1 and alpha so expect bugs.

**Comment by cliffordkleinsr** [comment]
Good start @JonathonRP. Looks like you already registered the package.

**Comment by cliffordkleinsr** [comment]
Hey @JonathonRP Some good news I've created a bunch of UI tests you can run using the motion-start local repo [here](https://github.com/cliffordkleinsr/motion-tests.git). You can do an `npm link` on the motion start  forked repo so you can test UI errors locally on the motion-test repository that I have shared

Some bad news is however that all the UI tests apart from the Animatepresence stack have failed. There might be some type errors with the Motion and MotionSSR components. There are also a lot of unused exports on the type definitions for these components

**Comment by JonathonRP** [comment]
> Good start @JonathonRP. Looks like you already registered the package. 

@cliffordkleinsr, thought I already registered the name motion-start with npm by publishing the package to npm

**Comment by cliffordkleinsr** [comment]
> > Good start @JonathonRP. Looks like you already registered the package.
> 
> I thought I already registered the name motion-start with npm by publishing the package to npm

No worries, went in to publish in the morning  then found out that it's taken. Hadn't read your comment.
😅

**Comment by JonathonRP** [comment]
> Hey @JonathonRP Some good news I've created a bunch of UI tests you can run using the motion-start local repo [here](https://github.com/cliffordkleinsr/motion-tests.git). You can do an `npm link` on the motion start  forked repo so you can test UI errors locally on the motion-test repository that I have shared
> 
> Some bad news is however that all the UI tests apart from the Animatepresence stack have failed. There might be some type errors with the Motion and MotionSSR components. There are also a lot of unused exports on the type definitions for these components
> 

@cliffordkleinsr, Interesting looking at this you're using svelte-motion and not motion-start

Maybe start a discord server so we can continue discussing there instead of this GitHub issue?

**Comment by cliffordkleinsr** [comment]
I tested with both. It's only worked for animatedpresence with the motion-start  library

**Comment by cliffordkleinsr** [comment]
Here is the discord btw:
EDIT : fresh link
 https://discord.gg/r6tSZjdR
 The library can be found here: https://www.npmjs.com/package/motion-start

**Comment by agrear** [comment]
### Good news!
[Framer Motion is now independent, introducing Motion](https://motion.dev/blog/framer-motion-is-now-independent-introducing-motion).

> For a long time, users have asked to use Framer Motion outside of React. Either with vanilla JS, or other frameworks like Vue.
>
> Today, Motion becomes more than a React library, with an amazing set of [vanilla APIs](https://motion.dev/docs/quick-start) that bring Motion's hybrid engine and extensive feature set to everyone.

**Comment by jbirtel** [comment]
@JonathonRP do you think your library is a good alternative for this OG svelte-motion library? Is it stable to use with Svelte v5?
I want to migrate to Svelte 5 in the project, but I need a replacement for svelte-motion. 
Can you please quickly explain what your repo is still lacking, if there is anything?

**Comment by JonathonRP** [comment]
> @JonathonRP do you think your library is a good alternative for this OG svelte-motion library? Is it stable to use with Svelte v5?
> I want to migrate to Svelte 5 in the project, but I need a replacement for svelte-motion. 
> Can you please quickly explain what your repo is still lacking, if there is anything?

Currently it lacks nothing from svelte-motion, and the main branch is stable for now. The only differences are I tried to stick closer to the framer-motion declarative API and some changes because of svelte 5 reactivity changes.

There is a discord community you can join from here
https://github.com/JonathonRP/motion-start
Feel free to ask any questions, I'm also there to help if any issues

I have future ideas for the lib, that could be breaking from current main and svelte-motion but I'll put it out as next for testing and then once stable most likely a major version.
